src/sidebar/services/groups.js

```diff
@@ -225,75 +234,68 @@ function groups(
           });
         }
         groupApiRequests = groupApiRequests.concat(selectedGroupApi);
-        return Promise.all(groupApiRequests).then(
-          ([myGroups, featuredGroups, token, selectedAnn, selectedGroup]) => {
-            // If there is a direct-linked group, add it to the featured groups list.
-            const allFeaturedGroups =
-              selectedGroup !== null &&
-              !featuredGroups.some(g => g.id === selectedGroup.id)
-                ? featuredGroups.concat([selectedGroup])
-                : featuredGroups;
-
-            // If there's a selected annotation it may require an extra api call
-            // to fetch its group.
-            if (selectedAnn) {
-              // Set the directLinkedAnnotationGroupId to be used later in
-              // the filterGroups method.
-              directLinkedAnnotationGroupId = selectedAnn.group;
-
-              const selectedAnnGroup = myGroups
-                .concat(allFeaturedGroups)
-                .some(g => g.id === selectedAnn.group);
-
-              // If the direct-linked annotation's group has not already been fetched,
-              // fetch it.
-              if (!selectedAnnGroup) {
-                return fetchGroup({
-                  id: selectedAnn.group,
-                  expand: params.expand,
-                }).then(directLinkedAnnGroup => {
-                  // If the directLinkedAnnotation's group fetch failed, return
-                  // the list of groups without it.
-                  if (!directLinkedAnnGroup) {
-                    return [
-                      combineGroups(myGroups, allFeaturedGroups, documentUri),
-                      token,
-                    ];
-                  }
-
-                  // If the directLinkedAnnotation's group fetch was successful,
-                  // combine it with the other groups.
-                  return [
-                    combineGroups(
-                      myGroups,
-                      allFeaturedGroups.concat(directLinkedAnnGroup),
-                      documentUri
-                    ),
-                    token,
-                  ];
-                });
+        return Promise.all(groupApiRequests);
+      })
+      .then(([myGroups, featuredGroups, token, selectedAnn, selectedGroup]) => {
+        // Step 3. Add the direct-linked group to the list of featured groups,
+        // and if there was a direct-linked annotation, fetch its group if we
+        // don't already have it.
+
+        // If there is a direct-linked group, add it to the featured groups list.
+        let allFeaturedGroups =
+          selectedGroup !== null &&
+          !featuredGroups.some(g => g.id === selectedGroup.id)
+            ? featuredGroups.concat([selectedGroup])
```

Comment:
If I'm reading this correctly, the direct-link annotation's group is at the end of the featured-groups list if it wasn't in the featured groups list to start with, but could be anywhere within it if it was already there (unless sorting happens later)? Is this intentional? Could it cause an inconsistency?

(edited: Looks like maybe the `combineGroups` stuff later takes care of sorting?)

Reply:
From a quick scan, it looks like groups are sorted by type in `groups.all` and later sorted by organization in `groupsByOrganization` when presenting groups. I can't see anything which sorts them in lexicographic order here. Up until now it looks like the client relied on the API response to sort groups by name and then by ID. So it looks like this could potentially cause the direct-linked group to be in non-alphabetical order in the store.

Since there is typically only one direct-linked group and it appears in its own "Currently viewing" section in the groups menu UI however, I don't think that will actually matter.